lesson7-questions.txt

```diff
@@ -67,14 +78,17 @@ Try that last line.  You will see that it creates a subscription object, but tha
 object is not entirely empty.  Which attribute is set? Where does it get that
 value?
 
+A subscription object is returned. The attribute that is set is the user_id attribute. Its automatically
+set because of its relationship defined in the User model.
 
 
 7. Suppose that user is an instance of User, and forum is an instance of Forum.  What
 Active Record Statement could you use to retrieve all the user's posts for that
 forum?  Hint: You need to have a where in your statement. Try this out in the console.
 You will see the SQL that is executed logged to the screen.  What is that SQL?
 
-
+user_posts = Post.where(user_id: user.id, forum_id: forum.id)
```

Comment:
Correct. Alternatively, you could make use of the association methods like so:

```ruby
user_posts = user.posts.where(forum: forum)
```

or

```ruby
user_posts = forum.posts.where(user: user)
```